### ISSUE TITLE
[Bug][A11y]: Menu item with href fails to activate using a screen reader 

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#4997]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Menu

### Expected behavior

Using a screen reader, a virtual click on a menu item with an href should activate the control and navigate to the new url.

### Actual behavior

Using VoiceOver on iOS, double tapping to activate a menu item with an href does not activate the control and fails to navigate to the new url.

I suspect this is because the Menu Item handles pointer and keyboard events when proxying the event to activate the link in the shadowDOM, yet it fails to handle the click event that VoiceOver triggers at the center of the element with VoiceOver cursor focus.

### Screenshots

The following screen recording demonstrates the issue using Accessibility Inspector to trigger a press event at on the element focused using the Accessibility Inspector. The virtual click event should be handled by the Menu item and the browser should navigate, but instead nothing happens.

https://github.com/user-attachments/assets/9d3a05df-68b7-46a9-acbd-d6b16bfb2453

### How can we reproduce this issue?

1. On iOS with VoiceOver on, go to https://opensource.adobe.com/spectrum-web-components/storybook/index.html?path=/story/menu-item--href
2. Swipe to move the VoiceOver cursor to the menu item labelled "Edit the Documentation Site"
3. Double tap to activate the control, which should open to the Spectrum Web Components documentation site
4. Nothing happens

### Severity

SEV 1
